Make it compile against Jellyfin

hawken93 opened the pull request:
Needs Jellyfin.Naming (Emby.Naming) to compile but it doesn't exist currently afaik

hawken93 committed:
Initial csproj Jellyfin update

hawken93 committed:
Port logging to Microsoft.Extensions.Logging

Bond-009 commented on Emby.AutoOrganize/Emby.AutoOrganize.csproj:
```diff
@@ -35,8 +35,8 @@
   <ItemGroup>
-    <PackageReference Include="MediaBrowser.Naming" Version="1.1.7-beta" />
-    <PackageReference Include="mediabrowser.server.core" Version="3.6.0.43-beta" />
+    <PackageReference Include="Jellyfin.Common" Version="10.0.1" />
```
Common is included in Controller

hawken93 replied:
"error CS0234: The type or namespace name 'Controller' does not exist in the namespace 'MediaBrowser' (are you missing an assembly reference?)"

this disappeared after I added that line...

hawken93 replied:
Never mind! :)

hawken93 commented:
https://github.com/jellyfin/jellyfin/pull/587

hawken93 commented:
I didn't rewrite the code fully yet, I expect compile errors when the packages are up but I will continue then :)

joshuaboniface commented:
@hawken93 Still a WIP?

hawken93 committed:
Remove thumb code, leave files behind

hawken93 committed:
BOM removal

hawken93 committed:
Make sure files end on newlines

hawken93 commented:
@joshuaboniface It's way closer to compiling now, but I need some help with the remaining ones.. I cleaned up the logging part and updated to compile against 10.1.0. @Bond-009 maybe I can ask for your help here

hawken93 commented:
Errors:

```
Core/FileOrganizationService.cs(148,25): error CS0117: 'EventHelper' does not contain a definition for 'FireEventIfNotNull' [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/FileOrganizationService.cs(199,25): error CS0117: 'EventHelper' does not contain a definition for 'FireEventIfNotNull' [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/FileOrganizationService.cs(205,25): error CS0117: 'EventHelper' does not contain a definition for 'FireEventIfNotNull' [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/FileOrganizationService.cs(284,29): error CS0117: 'EventHelper' does not contain a definition for 'FireEventIfNotNull' [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/FileOrganizationService.cs(288,29): error CS0117: 'EventHelper' does not contain a definition for 'FireEventIfNotNull' [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/FileOrganizationService.cs(306,25): error CS0117: 'EventHelper' does not contain a definition for 'FireEventIfNotNull' [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/EpisodeFileOrganizer.cs(315,61): error CS1503: Argument 1: cannot convert from 'MediaBrowser.Model.IO.IFileSystem' to 'MediaBrowser.Controller.Providers.IDirectoryService' [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/EpisodeFileOrganizer.cs(811,43): error CS1061: 'Series' does not contain a definition for 'InternalId' and no accessible extension method 'InternalId' accepting a first argument of type 'Series' could be found (are you missing a using directive or an assembly reference?) [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/EpisodeFileOrganizer.cs(945,35): error CS1061: 'Series' does not contain a definition for 'InternalId' and no accessible extension method 'InternalId' accepting a first argument of type 'Series' could be found (are you missing a using directive or an assembly reference?) [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
```
Warnings:
```
Data/SqliteFileOrganizationRepository.cs(95,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Data/SqliteFileOrganizationRepository.cs(118,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Data/SqliteFileOrganizationRepository.cs(134,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/MovieFileOrganizer.cs(273,28): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Core/EpisodeFileOrganizer.cs(459,28): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
Data/SqliteFileOrganizationRepository.cs(367,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/root/git/plugins-2-broken/jellyfin-plugin-autoorganize/Emby.AutoOrganize/Emby.AutoOrganize.csproj]
```

hawken93 commented:
This is going to be a big PR but I try my best to keep each commit very targeted, so I recommend to review commit by commit

hawken93 commented:
```
commit 391b48614d598b5fbfad68b610520a5ae7a57b70
Author: Bond_009 <bond.009@outlook.com>
Date:   Fri Dec 28 15:21:02 2018 +0100

    Remove FireEventIfNotNull
    
    It's a pretty useless "helper" class
```
:/ ok so how do I do this..

hawken93 committed:
Refactor out FireEventIfNotNull

hawken93 committed:
InternalId -> Id

hawken93 committed:
Add DirectoryService wrapper

hawken93 commented:
Compiling! Thanks @cvium!

hawken93 commented:
Decided to leave the warnings for the c# devs with skillz to sort out. It's not critical

hawken93 commented:
@joshuaboniface ping, I'm ready

cvium reviewed (approved):
A lot of the logger statements should be cleaned up, but I think it's good enough for now -- don't want you to get all depressed working on the plugins lol. We should have an issue to track the async warnings and general cleanup though.

hawken93 commented:
I want to get rid of the general exception catchers, but I know it will bre stuff
